src/content/en/showcase/2017/matterport2.md

```diff
@@ -0,0 +1,122 @@
+non-VR users. We also decided to prioritize support for the Daydream headset and controller in
+order to deliver a high quality experience for one device, rather than compromising the experience
+to accommodate a wider range of devices.
+
+## Switching to VR mode
+
+The easiest way we could think of for someone to enter VR was to detect when someone puts their
+phone in a VR headset while having our WebGL player open. This can be done by listening for the
+`vrdisplayactivate` event, which Chrome triggers when a user puts the phone in the Daydream headset
+and the NFC chip is detected.
+
+However, there is a small, but important difference between `vridisplayactivate` and other
+user-initiated events, e.g. onclick: Since we have to unload our normal WebGL player and load the
+WebVR player once we detect the user wanting to go into VR, we need a second or two of loading.
+Using `onclick` we are able to load this in the background during the controller calibration
+screen.  With `vrdisplayactivate`, though, we do not get to start loading until after the
+calibration, which ends up stalling the main thread when the user has the headset on, causing
+uncomfortable flickering:
+
+<a href="https://www.matterport.com/virtual-reality/webvr/"><img
+src="/web/showcase/2017/images/matterport/timeline.png" class="attemptfg-right"></a>
+
+Until `vrdisplayactivate` fires earlier, using `onclick` and a button provides users with the best
+experience for now.
+
+## Performance
+
+We knew performance would have to be rock solid for the experience to work well, so we started with
+the assumption that we would only implement a bare bones experience: no bells and whistles, just
+simple navigation inside a space.
+
+We quickly made a prototype with <a
+href="https://github.com/borismus/webvr-boilerplate">webvr-boilerplate</a> and a stripped down
+version of our loading and rendering code. In our first run we only got about 10-20 FPS: In our
+normal WebGL app, that would feel bad, but in VR it’s downright nauseating.  As we often do, we
+fired up the Chrome Developer Tools, and connected it to a Pixel phone. We got four major
+takeaways:
+
+- Know your rendering code!
+    - Webvr-boilerplate rendered the screen a total of four times, two for each eye. Removing one
+      extraneous line of code quickly halved our rendering time! This has since been fixed in the
+      newest version of webvr-boilerplate, but we suspect that this particular bug has caused a lot
+      of developers to get the wrong impression of webvr performance.
+- Profile on the actual device you are building for, not your developer computer
+    - Mobile devices and browsers have very different performance characteristics from your Mac;
+      make sure you’re optimizing the real bottleneck.
+- You don’t need all those pixels
+    - Be aware how many pixels you are rendering. In THREE.js, this is set with
```

Comment:
THREE.js is a very popular and well known framework for doing 3D graphics on the web. THREE.js is to WebVR what jQuery or React are to PWAs. Have reworded slightly anyway.